shared/js/l10n.js

```diff
@@ -15,46 +15,33 @@
   /* jshint browser:true */
 
   var io = {
-    load: function load(url, callback, sync) {
+
+    _load: function(type, url, callback, sync) {
       var xhr = new XMLHttpRequest();
+      var needParse;
 
       if (xhr.overrideMimeType) {
-        xhr.overrideMimeType('text/plain');
+        xhr.overrideMimeType(type);
       }
 
       xhr.open('GET', url, !sync);
 
-      xhr.addEventListener('load', function io_load(e) {
-        if (e.target.status === 200 || e.target.status === 0) {
-          callback(null, e.target.responseText);
+      if (type === 'application/json') {
+        //  Gecko 11.0+ forbids the use of the responseType attribute when
+        //  performing sync requests (NS_ERROR_DOM_INVALID_ACCESS_ERR).
+        //  We'll need to JSON.parse manually.
+        if (sync) {
+          needParse = true;
         } else {
-          callback(new L10nError('Not found: ' + url));
+          xhr.responseType = 'json';
         }
-      });
-      xhr.addEventListener('error', callback);
-      xhr.addEventListener('timeout', callback);
-
-      // the app: protocol throws on 404, see https://bugzil.la/827243
-      try {
-        xhr.send(null);
-      } catch (e) {
-        callback(new L10nError('Not found: ' + url));
-      }
-    },
-
-    loadJSON: function loadJSON(url, callback) {
-      var xhr = new XMLHttpRequest();
-
-      if (xhr.overrideMimeType) {
-        xhr.overrideMimeType('application/json');
       }
 
-      xhr.open('GET', url);
-
-      xhr.responseType = 'json';
-      xhr.addEventListener('load', function io_loadjson(e) {
+      xhr.addEventListener('load', function io_onload(e) {
         if (e.target.status === 200 || e.target.status === 0) {
-          callback(null, e.target.response);
+          callback(
+            null, needParse ?
+              JSON.parse(e.target.response) : e.target.response);
```

Comment:
It turns out Sinon'JS's FakeXHR doesn't expose the `response` property, so I had to go back to falling back onto `responseText` for tests.